graylog2-server/src/main/java/org/graylog2/indexer/MongoIndexSet.java

```diff
@@ -188,6 +181,14 @@ public Optional<Integer> extractIndexNumber(final String indexName) {
         }
     }
 
+    private int indexPrefixLength(String indexName) {
+        int length = config.indexPrefix().length() + 1;
+        if (indexName.contains(WARM_INDEX_INFIX)) {
```

Comment:
```suggestion
        if (indexName.contains("_" + WARM_INDEX_INFIX)) {
```
We could make this match more precise.
Also, could we add a validation that prevents the user from creating index sets with the name `warm_`?
Or are we sure this doesn't cause any problems?

Reply:
I just tested it. using my change fixes it for an index set with prefixes `warm_`
```
green open warm__warm_0 wM-hSUntTqqy8Go1WrOqeg 4 0 196 0  201kb  201kb
green open warm__2      KF3H7fBUTNaNezMJJDlIZA 5 0  13 0 13.6kb 13.6kb
green open warm__1      k9Fcst3QQIONOIZ3FPTY0A 5 0 959 0  1.1mb  1.1mb
```

Reply:
I added also the validation for the keyword `warm`